### ISSUE TITLE
Refactor SetInput/GetOutput methods (that are removed in VTK 6)

### ISSUE THREAD

**Issue by certik** [enthought/mayavi#74]
As of 0dedbd920f1b242ca7ecfaa4474bae5cfc79af69, when using VTK 6, mayavi builds fine, but we get runtime exceptions:
```
In [3]: from mayavi import mlab
No handlers could be found for logger "traits.has_traits"

In [4]: mlab.options.offscreen = True

In [5]: mlab.test_contour3d()
---------------------------------------------------------------------------
TraitError                                Traceback (most recent call last)
/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/messenger.pyc in send(obj, event, *args, **kw_args)
    306 
    307 def send(obj, event, *args, **kw_args):
--> 308     _messenger.send(obj, event, *args, **kw_args)
    309 send.__doc__ = _messenger.send.__doc__
    310 

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/messenger.pyc in send(self, source, event, *args, **kw_args)
    247                         inst = obj()
    248                         if inst:
--> 249                             getattr(inst, meth)(source, event, *args, **kw_args)
    250                         else:
    251                             # Oops, dead reference.

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/tvtk_base.pyc in update_traits(self, obj, event)
    454                     setattr(self, name, warn)
    455                 else:
--> 456                     setattr(self, name, val)
    457         # Reset the warning state.
    458         vtk.vtkObject.SetGlobalWarningDisplay(warn)

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/traits/mupe/lib/python2.7/site-packages/traits-4.3.0-py2.7-linux-x86_64.egg/traits/trait_handlers.pyc in error(self, object, name, value)
    167         """
    168         raise TraitError( object, name, self.full_info( object, name, value ),
--> 169                           value )
    170 
    171     def arg_error ( self, method, arg_num, object, name, value ):

TraitError: The 'alpha_bit_planes' trait of a RenderWindow instance must be 'f' or 'false' or 'n' or 'no' or 'off' or 'on' or 't' or 'true' or 'y' or 'yes' or 0 or 1, but a value of 8 <type 'int'> was specified.
---------------------------------------------------------------------------
TraitError                                Traceback (most recent call last)
<ipython-input-5-ba825038dacc> in <module>()
----> 1 mlab.test_contour3d()

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/tools/helper_functions.pyc in test_contour3d()
    492     scalars = x * x * 0.5 + y * y + z * z * 2.0
    493 
--> 494     obj = contour3d(scalars, contours=4, transparent=True)
    495     return obj
    496 

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/tools/helper_functions.pyc in the_function(*args, **kwargs)
     34 
     35     def the_function(*args, **kwargs):
---> 36         return pipeline(*args, **kwargs)
     37 
     38     if hasattr(pipeline, 'doc'):

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/tools/helper_functions.pyc in __call__(self, *args, **kwargs)
     74             scene = figure.scene
     75         else:
---> 76             scene = tools.gcf().scene
     77         if scene is not None:
     78             self._do_redraw = not scene.disable_render

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/tools/figure.pyc in gcf(engine)
    115     scene = engine.current_scene
    116     if scene is None:
--> 117         return figure(engine=engine)
    118     return scene
    119 

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/tools/figure.pyc in figure(figure, bgcolor, fgcolor, engine, size)
     67             __scene_number_list.update((name,))
     68             name = 'Mayavi Scene %d' % name
---> 69             engine.new_scene(name=name, size=size)
     70             engine.current_scene.name = name
     71         else:

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/apptools/wm2g/lib/python2.7/site-packages/apptools-4.2.0-py2.7.egg/apptools/scripting/recordable.pyc in _wrapper(*args, **kw)
     43         if not record:
     44             # If the method was not recorded, just call it.
---> 45             result = func(*args, **kw)
     46 
     47         return result

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/core/engine.pyc in new_scene(self, viewer, name, **kwargs)
    432                     factory_kwargs[arg] = value
    433 
--> 434             viewer = self.scene_factory(**factory_kwargs)
    435             process_ui_events()
    436 

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/mayavi/core/off_screen_engine.pyc in off_screen_viewer_factory(size)
     14 def off_screen_viewer_factory(size=(400, 350)):
     15     """A factory that creates an offscreen viewer."""
---> 16     win = TVTKWindow(off_screen_rendering=True)
     17     # Set all preferences.
     18     set_scene_preferences(win.scene)

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/pyface/tvtk_scene.pyc in __init__(self, **traits)
    917         """All the keyword arguments are passed on to the `TVTKScene`
    918         instance created."""
--> 919         self.scene = TVTKScene(**traits)

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/pyface/tvtk_scene.pyc in __init__(self, parent, **traits)
    199         self._def_pos = 1
    200 
--> 201         self.control = self._create_control(parent)
    202         self._renwin.update_traits()
    203 

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/pyface/tvtk_scene.pyc in _create_control(self, parent)
    803         self._interactor.initialize()
    804         self._interactor.render()
--> 805         self.light_manager = light_manager.LightManager(self)
    806 
    807         if self.off_screen_rendering:

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/pyface/light_manager.pyc in __init__(self, renwin, **traits)
    349         self.lights = []
    350         for i in range(self.number_of_lights):
--> 351             light = CameraLight(self.renwin)
    352             self.lights.append(light)
    353 

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/pyface/light_manager.pyc in __init__(self, renwin, **traits)
    165     #################################################################
    166     def __init__(self, renwin, **traits):
--> 167         self.glyph = LightGlyph()
    168         super(CameraLight, self).__init__(**traits)
    169         self.source.light_type = 'camera_light'

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/mayavi/toug/lib/python2.7/site-packages/mayavi-4.3.0-py2.7-linux-x86_64.egg/tvtk/pyface/light_manager.pyc in __init__(self)
     57         t.rotate_y(90.0)
     58         t.translate((-2, 0, 0))
---> 59         tf.input = arrow.output
     60 
     61         mapper = tvtk.PolyDataMapper()

/auto/nest/nest/u/ondrej/repos/python-hpcmp2/opt/traits/mupe/lib/python2.7/site-packages/traits-4.3.0-py2.7-linux-x86_64.egg/traits/trait_handlers.pyc in _read_only(object, name, value)
     99 def _read_only ( object, name, value ):
    100     raise TraitError, "The '%s' trait of %s instance is 'read only'." % (
--> 101                       name, class_of( object ) )
    102 
    103 def _undefined_get ( object, name ):

TraitError: The 'input' trait of a TransformFilter instance is 'read only'.

In [6]: 
```
And the reason is that Mayavi still uses the old SetInput/GetOutput methods which are no longer available in VTK 6.

@prabhuramachandran, do you know what the right fix should be? I have run into exactly this issue before with some C++ examples and the fix turned out to be straightforward:

http://www.vtk.org/Wiki/VTK/VTK_6_Migration/Replacement_of_SetInput

Pretty much, one just has to use either `SetInputConnection` or `SetInputData` depending on whether one is chaining filters or not. In Mayavi, should this be fixed in the TVTK module?

**Comment by jamasi** [comment]
I'm also affected by this, as sabayon/gentoo are also shipping vtk-6(.1) now. When can we expect a solution to this issue?

**Comment by jdmarch** [comment]
@jamasi Imminent, see https://github.com/enthought/mayavi/blob/master/docs/CHANGES.txt

**Comment by dmsurti** [comment]
Fixed as of Mayavi 4.4.2 and VTK 6.2. Closed.
```
In [1]: from mayavi import mlab

In [2]: mlab.options.offscreen = True

In [3]: mlab.test_contour3d()
Out[3]: <mayavi.modules.iso_surface.IsoSurface at 0x126e842f0> 
```

**Comment by certik** [comment]
@dmsurti great job, thanks!
